tasks/run-local.sh

```diff
@@ -169,15 +170,14 @@ if [ -n "$PR" ]; then
 
     podman exec -i cockpituous-tasks sh -exc "
     cd bots;
-    ./tests-trigger -f --repo $PR_REPO $PR unit-tests;
+    ./tests-scan -p $PR --amqp 'localhost:5671' --repo $PR_REPO;
```

Comment:
Does it make sense in this order? To do tests-scan, then do loop waiting for status to appear (the one, we already hope to have scanned) and then check if the original scan inserted it into queue?
Or am I reading this wrongly?

Reply:
I'm confused as to where the test is triggered now? Does `COCKPIT_TESTMAP_INJECT` mean you can run tests-scan and it'll always return the value ? If so it kinda feels like using less of the bots in the process and introducing a hack :/

Reply:
The previous tests-trigger was only a crutch because we had the `unit-tests` context in `_manual`, so that a direct tests-scan wouldn't see it. Now I think it makes sense -- we call tests-scan (with the injected context from the environment), which will trigger "unit-tests" on github *and* queue it. The retry loop below is read-only (`-d`), it just waits until tests-scan finished its thing and the status is visible on github. We could probably also drop it and just leave it to the "waiting for unit-tests run to finish..." loop, but I suppose Sanne's intention was to make it clearer in case of a failure *why* it failed -- was it not queued or never ran from the queue?

Or I entirely misunderstood your question.

Reply:
@Gundersanne: See https://github.com/cockpit-project/bots/pull/1623 and my initial attempt at https://github.com/cockpit-project/bots/pull/1622 ; I just can't see how to make this work with tests-trigger and manual for bots (and that approach is totally not what our infra is doing). We need this little bit of a dynamic testmap for this purpose. Of course I'd be  happy to have a better approach.

Reply:
I think i understand it now, +1